internal/storage/sql_provider.go

```diff
@@ -219,7 +218,7 @@ func (p *SQLProvider) LoadUserInfo(ctx context.Context, username string) (info m
 	case err == nil:
 		return info, nil
 	case errors.Is(err, sql.ErrNoRows):
-		if _, err = p.db.ExecContext(ctx, p.sqlUpsertPreferred2FAMethod, username, authentication.PossibleMethods[0]); err != nil {
+		if _, err = p.db.ExecContext(ctx, p.sqlUpsertPreferred2FAMethod, username, ""); err != nil {
```

Comment:
Upserting in a method called LoadUserInfo seems broken to me. This relates to previous comments.

Reply:
Yeah  I understand,the reason this was done was to prevent the methods used to check which 2FA technologies the user had registered each had their own method. Adding a single LoadUserInfo method allows us to reduce the number of methods by 3, and execute 1 query instead of 4 in the instance of the preferred method existing and 3 instead of 4 in the instance of the preferred method not existing (first time).